packages/react/tasks/build.js

```diff
@@ -132,8 +140,17 @@ function getRollupConfig(input, outDir, useTS) {
       commonjs({
         include: /node_modules/,
       }),
-      // Modify plugins for builds that require typescript
-      ...(useTS ? getTSPlugins(outDir) : getPlugins()),
+      typescript({
+        noEmitOnError: true,
+        noForceEmit: true,
+        outputToFilesystem: false,
+        compilerOptions: {
+          emitDeclarationOnly: true,
+          rootDir,
+          outDir,
+        },
+      }),
+      babel(babelConfig),
```

Comment:
Some of these typescript config values seems to exist across a few different files (perhaps in slightly different forms). Is it possible to cue off of tsconfig files for these instead? Or is this maybe a limitation of the rollup tooling that we can't easily do that type of thing?

Reply:
@jdharvey-ibm yeah, this is problematic... We have (at least) 2 different kinds of TS configs:
1. The `tsconfig.json` files are mainly used in development environment, e.g.: by IDE, eslint, etc.
2. The inline rollup configs, which emit to build directories.

In this PR, I didn't quite think about the difference between group 1 and group 2 - I just based on whatever was there for each group. I guess unifying into one base config and then extending with options that absolutely must be different (e.g. rootDir/outDir) is the right way to go. I'll look at this together with addressing your other comment (https://github.com/carbon-design-system/carbon/pull/14714#discussion_r1341477056).